nodepool/nodepool-config.yml

```diff
@@ -0,0 +1,89 @@
+---
+# Documentation reference:
+# https://docs.openstack.org/infra/nodepool/feature/zuulv3/configuration.html
+
+images-dir: /opt/nodepool/images
+
+zookeeper-servers:
+  - host: localhost
+    port: 2181
+
+labels:
+  - name: ubuntu-xenial
+    # TODO(odyssey4me):
+    # This is a temporarily low setting as the nodes will
+    # not be used until the jenkins integration is complete.
+    # It is here purely to validate that everything is
+    # working as it should be.
+    min-ready: 3
+    # TODO(odyssey4me):
+    # This is purely here for test purposes to make sure any
+    # existing nodes that haven't been used will get recreated
+    # every hour.
+    max-ready-age: 3600
+
+providers:
+  - name: pubcloud-iad
+    region-name: 'IAD'
+    cloud: public_cloud
+    api-timeout: 60
+    boot-timeout: 120
+    hostname-format: nodepool-{label.name}-{provider.name}-{node.id}
+    image-name-format: nodepool-{image_name}-{timestamp}
+    diskimages: &provider_diskimage_block
+      - name: ubuntu-xenial
+        config-drive: true
+    pools: &provider_pools_block
+      - name: main
+        max-servers: 2
+        availability-zones: []
+        labels:
+          - name: ubuntu-xenial
+            diskimage: ubuntu-xenial
+            flavor-name: general1-8
```

Comment:
How do I overwrite this? Do I have to create another pool with a different flavour?

Reply:
There are bits here to provide combinations:

label - an arbitrary name (could be ubuntu-xenial-big or ubuntu-xenial-metal)
diskimage - defines the image to use (which effectively defined the contents)
flavor-name - for this provider, the flavor that goes with this label... for others it might be different (ie pubcloud and phobos may use different flavor names)

There are also multiple labels per pool, and multiple pools per provider. Our jenkins integration would then request a particular label from a particular pool for the job. So what you do depends on what you're looking to achieve. If you just want access to more flavor profiles, then we create more labels in the same pool. If you want to perhaps use the same label, but have different flavor profiles per pool then I think that's doable.